util/perftest.c

```diff
@@ -0,0 +1,116 @@
+#include <assert.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "monoclock.h"
+#include "warnp.h"
+
+#include "perftest.h"
+
+/**
+ * perftest_buffers(num_bytes, sizes, num_sizes, num_warm_up, init_func, func,
+ *     cookie):
+ * Time using ${func} to process ${num_bytes} bytes in blocks of ${sizes}.
+ * Before timing any block sizes, process ${num_warm_up} blocks of the maximum
```

Comment:
I think it probably makes more sense to have `num_bytes_warmup` bytes rather than `num_warm_up` blocks.

Also, I would call the parameters `nbytes`, `nsizes`, and `nbytes_warmup`.

Reply:
Ok.  At the moment, we have:
```
static int
perftest_func(void * cookie, uint8_t * buf, size_t buflen, size_t num_buffers)
```
in `tests/`.  Should I rename that to `nbuffers`?

Reply:
I would probably go with `nbufs`.

Reply:
Or possibly `nreps` since it's not the number of buffers but rather the number of times we should repeat on the same buffer, yes?